homeassistant/components/group/entity.py

```diff
@@ -287,22 +290,29 @@ def _set_tracked(self, entity_ids: Collection[str] | None) -> None:
         if not entity_ids:
             self.tracking = ()
             self.trackable = ()
+            self.single_active_domain = None
             return
 
         registry: GroupIntegrationRegistry = self.hass.data[REG_KEY]
         excluded_domains = registry.exclude_domains
 
         tracking: list[str] = []
         trackable: list[str] = []
+        active_domains: set[str] = set()
         for ent_id in entity_ids:
             ent_id_lower = ent_id.lower()
             domain = split_entity_id(ent_id_lower)[0]
+            active_domains.add(domain)
             tracking.append(ent_id_lower)
             if domain not in excluded_domains:
```

Comment:
Excluded domains probably shouldn't be ignored here as well.
```suggestion
            tracking.append(ent_id_lower)
            if domain not in excluded_domains:
                active_domains.add(domain)
```

Reply:
Applying this breaks the tests

Reply:
I am not safe in further refactoring this. May be that is good for a follow up PR?

Reply:
This is not a refactoring, this is a potential bug. If domain is in `excluded_domains`, it shouldn't be "active". Unless there is something I'm missing.

Reply:
Fixed it